app/src/main/java/bitcoin/wallet/modules/fulltransactioninfo/FullTransactionInfoInteractor.kt

```diff
@@ -29,19 +30,33 @@ class FullTransactionInfoInteractor(private val adapter: IAdapter?, private val
         transaction?.let {
             val viewItem = getTransactionRecordViewItem(it, adapter)
             transactionRecordViewItem = viewItem
+            fetchExchangeRate(transaction)
             delegate?.didGetTransactionInfo(viewItem)
         }
     }
 
-    private fun getTransactionRecordViewItem(record: TransactionRecordNew, adapter: IAdapter): TransactionRecordViewItem {
-        val rates = exchangeRateManager.exchangeRates
-        val convertedValue = rates[adapter.coin.code]?.let { it * record.amount }
+    private fun fetchExchangeRate(transaction: TransactionRecord) {
+        transaction.timestamp?.let { timestamp ->
+            ExchangeRateManager.getRate(coinCode = transaction.coinCode, currency = DollarCurrency().code, timestamp = timestamp)
```

Comment:
Why manager class is used directly inside interactor? I'd suggest to use pass exchangeRateManager when initializing, and through interface.